### ISSUE TITLE
Use hard breaks in formal grammar

### ISSUE THREAD

**Issue by amartini51** [apple/swift-book#54]
### Description

Currently, DocC doesn't render hard breaks: <https://github.com/apple/swift-docc/issues/477>

When support is added, we should change the formal grammar here to use blank lines between groups and hard breaks between definitions.

Currently, as a workaround to preserve this information, each definition is followed by a blank line in source, to make it appear on its own line in output, and groups are separated by two blank lines.  So something like this:

```md
> Grammar of a self expression:
>
> *self-expression* → **`self`** | *self-method-expression* | *self-subscript-expression* | *self-initializer-expression*
>
>
>
> *self-method-expression* → **`self`** **`.`** *identifier*
>
> *self-subscript-expression* → **`self`** **`[`** *function-call-argument-list* **`]`**
>
> *self-initializer-expression* → **`self`** **`.`** **`init`**
```

Should turn into something like this:

```md
> Grammar of a self expression:
>
> *self-expression* → **`self`** | *self-method-expression* | *self-subscript-expression* | *self-initializer-expression*
>
> *self-method-expression* → **`self`** **`.`** *identifier* \
> *self-subscript-expression* → **`self`** **`[`** *function-call-argument-list* **`]`** \
> *self-initializer-expression* → **`self`** **`.`** **`init`**
```

**Comment by Kyle-Ye** [comment]
The issue is fixed upstream on Swift 5.9 release. Can we revisit and fix it now? @amartini51 

You can test it via a later nightly toolchain of Swift 5.9 or wait for the next Xcode 15 beta.

**Comment by amartini51** [comment]
@Kyle-Ye I was excited to see support for hard breaks yesterday.  It took me two tries to write a script that transformed the TSPL grammar.  (The first script in AWK was easy to write, but couldn't add the ` \` at the end of lines, so I redid it this morning in Python.)

If you'd like to review the changes in the diff, please have a look at #155.  I also want to review the changes in a rendered HTML version, but I'm waiting for another PR to finish so I can re-use my GitHub Pages to host this set of changes for review.
